pgx/src/datum/array.rs

```diff
@@ -10,12 +10,13 @@ Use of this source code is governed by the MIT license that can be found in the
 use crate::{array::RawArray, layout::*, pg_sys, FromDatum, IntoDatum, PgMemoryContexts};
 use bitvec::slice::BitSlice;
 use core::ptr::NonNull;
+use pgx_utils::sql_entity_graph::metadata::{
+    ArgumentError, ReturnVariant, ReturnVariantError, SqlTranslatable, SqlVariant,
+};
```

Comment:
At first I was like "eh it's imported everywhere, sure: because it needs to be implemented everywhere".

Then I started wondering: Is this an inappropriate leakage/coupling of implementation details?

Then I remembered that this is no more peculiar than a Rust struct requiring Debug in order to be `format!("{:?}", debug_formatted)`.